tests/testthat/test-nest.R

```diff
@@ -311,73 +311,78 @@ test_that("unnest keeps list cols", {
 
 test_that("warn about old style interface", {
   df <- tibble(x = c(1, 1, 1), y = 1:3)
-  expect_warning(out <- nest(df, y), "data = c(y)", fixed = TRUE)
+  expect_snapshot((expect_warning(out <- nest(df, y))))
   expect_named(out, c("x", "data"))
 })
 
 test_that("only warn about unnamed inputs (#1175)", {
   df <- tibble(x = 1:3, y = 1:3, z = 1:3)
-  expect_warning(out <- nest(df, x, y, foo = z), "data = c(x, y)", fixed = TRUE)
+  expect_snapshot((expect_warning(out <- nest(df, x, y, foo = z))))
   expect_named(out, c("foo", "data"))
 })
 
 test_that("unnamed expressions are kept in the warning", {
   df <- tibble(x = 1:3, z = 1:3)
-  expect_warning(out <- nest(df, x, starts_with("z")), 'data = c(x, starts_with("z"))', fixed = TRUE)
+  expect_snapshot((expect_warning(out <- nest(df, x, starts_with("z")))))
   expect_named(out, "data")
 })
 
 test_that("can control output column name", {
   df <- tibble(x = c(1, 1, 1), y = 1:3)
-  expect_warning(out <- nest(df, y, .key = "y"), "y = c(y)", fixed = TRUE)
+  expect_snapshot((expect_warning(out <- nest(df, y, .key = "y"))))
   expect_named(out, c("x", "y"))
 })
 
 test_that("can control output column name when nested", {
   df <- dplyr::group_by(tibble(x = c(1, 1, 1), y = 1:3), x)
-  expect_warning(out <- nest(df, .key = "y"), "`.key`", fixed = TRUE)
+  expect_snapshot((expect_warning(out <- nest(df, .key = "y"))))
   expect_named(out, c("x", "y"))
 })
 
 test_that(".key gets warning with new interface", {
   df <- tibble(x = c(1, 1, 1), y = 1:3)
-  expect_warning(out <- nest(df, y = y, .key = "y"), ".key", fixed = TRUE)
+  expect_snapshot((expect_warning(out <- nest(df, y = y, .key = "y"))))
   expect_named(df, c("x", "y"))
 })
 
 test_that("cols must go in cols", {
   df <- tibble(x = list(3, 4), y = list("a", "b"))
-  expect_warning(unnest(df, x, y), "c(x, y)", fixed = TRUE)
+  expect_snapshot((expect_warning(unnest(df, x, y))))
 })
 
 test_that("need supply column names", {
   df <- tibble(x = 1:2, y = list("a", "b"))
-  expect_warning(unnest(df), "c(y)", fixed = TRUE)
+  expect_snapshot((expect_warning(unnest(df))))
 })
 
 test_that("sep combines column names", {
+  local_options(lifecycle_verbosity = "warning")
   df <- tibble(x = list(tibble(x = 1)), y = list(tibble(x = 1)))
-  expect_warning(out <- df %>% unnest(c(x, y), .sep = "_"), "names_sep")
+  expect_snapshot((expect_warning(out <- df %>% unnest(c(x, y), .sep = "_"))))
   expect_named(out, c("x_x", "y_x"))
 })
 
 test_that("unnest has mutate semantics", {
   df <- tibble(x = 1:3, y = list(1, 2:3, 4))
-  expect_warning(out <- df %>% unnest(z = map(y, `+`, 1)), "mutate")
+  expect_snapshot((expect_warning(out <- df %>% unnest(z = map(y, `+`, 1)))))
   expect_equal(out$z, 2:5)
 })
 
 test_that(".drop and .preserve are deprecated", {
+  local_options(lifecycle_verbosity = "warning")
```

Comment:
Added this since these are lifecycle warnings that are triggered once per session